e2e/perf/generate_perf_report.py

```
@@ -0,0 +1,438 @@
+                return f"{start.strftime('%m/%d/%Y')}"
+            else:
+                return f"{start.strftime('%m/%d')} - {end.strftime('%m/%d')}"
+        except:
+            return "-"
+
+    def post_github_comment(self, report_content: str, pr_number: str) -> bool:
+        """Post report as GitHub comment using gh CLI."""
+        try:
+            # Use gh CLI to post comment
+            cmd = ['gh', 'pr', 'comment', pr_number, '--body', report_content]
+            result = subprocess.run(cmd, capture_output=True, text=True)
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)